### ISSUE TITLE
https://wpt.fyi/api/runs?max-count=100 has duplicate entries for some runs

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt.fyi#54]
Moved from https://github.com/web-platform-tests/results-collection/issues/528, examples of duplication there.

After @Hexcles results receiver is in place, this will be a wpt.fyi issue. It's possible we should just accept duplicate runs, but if so we need to store all of them and have a bucket structure that allows for it.

**Comment by lukebjerring** [comment]
AFAIK, this issue is resolved by the way that we use an unique ID for the bucket that the report gets stored in. @Hexcles  is that true?

**Comment by Hexcles** [comment]
@lukebjerring right, we had a discussion earlier. File paths in GCS buckets are the problem.

Currently, the file paths are deterministic, constituted of SHAs and platform IDs. If we have duplicate runs from one or several runners for the same SHA and platform ID, we would currently overwrite the JSONs in the GCS buckets, but create *new* `TestRun` entries in Datastore (the primary key of `TestRun` is a unique ID, which isn't currently used anywhere).

This should be fixed by adding a a short random string to the file paths.
